app/javascript/controllers/categories_controller.js

```diff
@@ -0,0 +1,43 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="categories"
+export default class extends Controller {
+    static targets = ['chips']
+    static values = { categories: Array}
+    check(event){
+        const input = event.currentTarget.querySelector('input')
+        const parents = this.#nest_children_inside_parents()
+        if(input.value in parents){
+            const allInputs = this.chipsTarget.querySelectorAll('input')
+            const children = parents[input.value]
+            allInputs.forEach(i => {
+                if(children.includes(i.value)){
+                    if(input.checked){
+                        i.checked = true;
+                        i.dispatchEvent(new Event('change', { bubbles: true }));
+                    } else {
+                        i.checked = false;
+                        i.dispatchEvent(new Event('change', { bubbles: true }));
+                    }
+                    
+                }
+            });
+        }
+    }
+    #nest_children_inside_parents(){
```

Comment:
I did a mistake,
when I select a category I should select the parent of it, but when I deselect the parent I should deselect the children also